File: src/main/java/com/flowcommerce/core/storage/member/MemberEntity.java
package com.flowcommerce.core.storage.member;

import com.flowcommerce.core.domain.member.AuthProvider;
import com.flowcommerce.core.domain.member.MemberStatus;
import com.flowcommerce.core.storage.BaseEntity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@Table(name = "members")
@NoArgsConstructor(access = lombok.AccessLevel.PROTECTED)
@AllArgsConstructor
public class MemberEntity extends BaseEntity {

    @Column(unique = true, nullable = false)
    private String email;

    private String password;

    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, name = "member_status")
    private MemberStatus memberStatus;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AuthProvider authProvider;

    private String providerId;

    public static MemberEntity forLocalSignUp(String email, String encodedPassword, String name) {
        return new MemberEntity(
                email,
                encodedPassword,
                name,
                MemberStatus.ACTIVE,
                AuthProvider.LOCAL,
                null
        );
    }
}

File: src/main/kotlin/com/flowcommerce/core/storage/member/MemberEntity.kt
package com.flowcommerce.core.storage.member

import com.flowcommerce.core.domain.member.AuthProvider
import com.flowcommerce.core.domain.member.MemberStatus
import com.flowcommerce.core.storage.BaseEntity
import jakarta.persistence.*

@Entity
@Table(name = "members")
class MemberEntity protected constructor() : BaseEntity() {

    @Column(unique = true, nullable = false)
    var email: String = ""
        protected set

    var password: String? = null
        protected set

    var name: String? = null
        protected set

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, name = "member_status")
    var memberStatus: MemberStatus = MemberStatus.ACTIVE
        protected set

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var authProvider: AuthProvider = AuthProvider.LOCAL
        protected set

    var providerId: String? = null
        protected set

    companion object {
        fun forLocalSignUp(email: String, encodedPassword: String, name: String): MemberEntity {
            return MemberEntity().apply {
                this.email = email
                this.password = encodedPassword
                this.name = name
                this.memberStatus = MemberStatus.ACTIVE
                this.authProvider = AuthProvider.LOCAL
            }
        }
    }
}
